File: android-tappxi/src/cmdf2/tappxi/model/bean/Address.java
package cmdf2.tappxi.model.bean;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Collection;

import org.apache.http.NameValuePair;
import org.apache.http.message.BasicNameValuePair;
import org.json.JSONException;
import org.json.JSONObject;

import com.google.android.maps.GeoPoint;

public class Address implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	private int id;
	private String street;
	private String settlement;
	private String city;
	private String state;
	private String zipCode;
	private int latitude;
	private int longitude;
	
	private String reference;

	public Address(int id, String street, String settlement, String city,
			String state, String zipCode, GeoPoint geoPoint, String reference) {
		super();
		this.id = id;
		this.street = street;
		this.settlement = settlement;
		this.city = city;
		this.state = state;
		this.zipCode = zipCode;
		this.latitude = geoPoint.getLatitudeE6();
		this.longitude = geoPoint.getLongitudeE6();
		this.reference = reference;
	}

	public Address(String street, String settlement, String city, String state,
			String zipCode, GeoPoint geoPoint, String reference) {
		this(0, street, settlement, city, state, zipCode, geoPoint, reference);
	}

	public Address(String street, GeoPoint geoPoint, String reference) {
		this(0, street, "", "", "", "", geoPoint, reference);
	}

	public Address() {
		this(0, "", "", "", "", "", new GeoPoint(0, 0), "");
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getStreet() {
		return (street != null ? street : "").trim();
	}

	public void setStreet(String street) {
		this.street = street;
	}

	public String getSettlement() {
		return (settlement != null ? settlement : "").trim();
	}

	public void setSettlement(String settlement) {
		this.settlement = settlement;
	}

	public String getCity() {
		return (city != null ? city : "").trim();
	}

	public void setCity(String city) {
		this.city = city;
	}

	public String getState() {
		return (state != null ? state : "").trim();
	}

	public void setState(String state) {
		this.state = state;
	}

	public String getZipCode() {
		return (zipCode != null ? zipCode : "").trim();
	}

	public void setZipCode(String zipCode) {
		this.zipCode = zipCode;
	}

	public GeoPoint getGeoPoint() {
		return new GeoPoint(latitude, longitude);
	}

	public void setGeoPoint(GeoPoint geoPoint) {
		this.latitude = geoPoint.getLatitudeE6();
		this.longitude = geoPoint.getLongitudeE6();
	}

	public String toString() {
		return getStreet() + ", " + getSettlement() + ", " + getCity() + ", "
				+ getState() + ", " + getZipCode();
	}

	public Collection<NameValuePair> getNameValuePairs(String prefix) {
		Collection<NameValuePair> collection = new ArrayList<NameValuePair>();
		collection.add(new BasicNameValuePair(prefix + "street", getStreet()));
		collection.add(new BasicNameValuePair(prefix + "settlement",
				getSettlement()));
		collection.add(new BasicNameValuePair(prefix + "city", getCity()));
		collection.add(new BasicNameValuePair(prefix + "state", getState()));
		collection
				.add(new BasicNameValuePair(prefix + "zip_code", getZipCode()));
		collection.add(new BasicNameValuePair(prefix + "latitude", String
				.valueOf(latitude)));
		collection.add(new BasicNameValuePair(prefix + "longitude", String
				.valueOf(longitude)));
		collection.add(new BasicNameValuePair(prefix + "reference", getReference()));

		return collection;
	}

	public Collection<NameValuePair> getNameValuePairs() {
		return getNameValuePairs("");
	}

	public String getReference() {
		return reference;
	}

	public void setReference(String reference) {
		this.reference = reference;
	}

	public static Address fromJSONObject(JSONObject object) throws JSONException {
		return new Address(object.getInt("id"), object.getString("street"),
				object.getString("settlement"), object.getString("city"),
				object.getString("state"), object.getString("zip_code"),
				new GeoPoint(object.getInt("latitude"),
						object.getInt("longitude")), object.getString("reference"));
	}

}

File: src/Tappxi/Bundle/ApiBundle/Entity/Address.php
<?php

namespace Tappxi\Bundle\ApiBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Tappxi\Bundle\ApiBundle\Entity\Address
 *
 * @ORM\Table(name="address")
 * @ORM\Entity
 */
class Address
{
    /**
     * @var integer $id
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string $street
     *
     * @ORM\Column(name="street", type="string", length=90, nullable=false)
     */
    private $street;

    /**
     * @var string $settlement
     *
     * @ORM\Column(name="settlement", type="string", length=90, nullable=false)
     */
    private $settlement;

    /**
     * @var string $city
     *
     * @ORM\Column(name="city", type="string", length=90, nullable=false)
     */
    private $city;

    /**
     * @var string $state
     *
     * @ORM\Column(name="state", type="string", length=45, nullable=false)
     */
    private $state;

    /**
     * @var string $zipCode
     *
     * @ORM\Column(name="zip_code", type="string", length=5, nullable=false)
     */
    private $zipCode;

    /**
     * @var float $latitude
     *
     * @ORM\Column(name="latitude", type="float", nullable=false)
     */
    private $latitude;

    /**
     * @var float $longitude
     *
     * @ORM\Column(name="longitude", type="float", nullable=false)
     */
    private $longitude;



    /**
     * Get id
     *
     * @return integer
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set street
     *
     * @param string $street
     * @return Address
     */
    public function setStreet($street)
    {
        $this->street = $street;

        return $this;
    }

    /**
     * Get street
     *
     * @return string
     */
    public function getStreet()
    {
        return $this->street;
    }

    /**
     * Set settlement
     *
     * @param string $settlement
     * @return Address
     */
    public function setSettlement($settlement)
    {
        $this->settlement = $settlement;

        return $this;
    }

    /**
     * Get settlement
     *
     * @return string
     */
    public function getSettlement()
    {
        return $this->settlement;
    }

    /**
     * Set city
     *
     * @param string $city
     * @return Address
     */
    public function setCity($city)
    {
        $this->city = $city;

        return $this;
    }

    /**
     * Get city
     *
     * @return string
     */
    public function getCity()
    {
        return $this->city;
    }

    /**
     * Set state
     *
     * @param string $state
     * @return Address
     */
    public function setState($state)
    {
        $this->state = $state;

        return $this;
    }

    /**
     * Get state
     *
     * @return string
     */
    public function getState()
    {
        return $this->state;
    }

    /**
     * Set zipCode
     *
     * @param string $zipCode
     * @return Address
     */
    public function setZipCode($zipCode)
    {
        $this->zipCode = $zipCode;

        return $this;
    }

    /**
     * Get zipCode
     *
     * @return string
     */
    public function getZipCode()
    {
        return $this->zipCode;
    }

    /**
     * Set latitude
     *
     * @param float $latitude
     * @return Address
     */
    public function setLatitude($latitude)
    {
        $this->latitude = $latitude;

        return $this;
    }

    /**
     * Get latitude
     *
     * @return float
     */
    public function getLatitude()
    {
        return $this->latitude;
    }

    /**
     * Set longitude
     *
     * @param float $longitude
     * @return Address
     */
    public function setLongitude($longitude)
    {
        $this->longitude = $longitude;

        return $this;
    }

    /**
     * Get longitude
     *
     * @return float
     */
    public function getLongitude()
    {
        return $this->longitude;
    }

    public function toArray(){
        return array(
            'id' => $this->getId(),
            'street' => $this->getStreet(),
            'settlement' => $this->getSettlement(),
            'city' => $this->getCity(),
            'state' => $this->getState(),
            'zip_code' => $this->getZipCode(),
            'latitude' => $this->getLatitude(),
            'longitude' => $this->getLongitude(),
        );
    }

    public static function fromArray($array){
        $address = new Address();
        $address->setStreet($array['street']);
        $address->setSettlement($array['settlement']);
        $address->setCity($array['city']);
        $address->setState($array['state']);
        $address->setZipCode($array['zip_code']);
        $address->setLatitude($array['latitude']);
        $address->setLongitude($array['longitude']);
        return $address;
    }

    public function toJson(){
        return json_encode($this->toArray());
    }
}